// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message NetworkservicesBetaTcpRoute {
  string name = 1;
  string create_time = 2;
  string update_time = 3;
  string description = 4;
  repeated NetworkservicesBetaTcpRouteRules rules = 5;
  repeated string meshes = 6;
  repeated string gateways = 7;
  map<string, string> labels = 8;
  string project = 9;
  string location = 10;
  string self_link = 11;
}

message NetworkservicesBetaTcpRouteRules {
  repeated NetworkservicesBetaTcpRouteRulesMatches matches = 1;
  NetworkservicesBetaTcpRouteRulesAction action = 2;
}

message NetworkservicesBetaTcpRouteRulesMatches {
  string address = 1;
  string port = 2;
}

message NetworkservicesBetaTcpRouteRulesAction {
  repeated NetworkservicesBetaTcpRouteRulesActionDestinations destinations = 1;
  bool original_destination = 2;
}

message NetworkservicesBetaTcpRouteRulesActionDestinations {
  int64 weight = 1;
  string service_name = 2;
}

message ApplyNetworkservicesBetaTcpRouteRequest {
  NetworkservicesBetaTcpRoute resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteNetworkservicesBetaTcpRouteRequest {
  string service_account_file = 1;
  NetworkservicesBetaTcpRoute resource = 2;
}

message ListNetworkservicesBetaTcpRouteRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListNetworkservicesBetaTcpRouteResponse {
  repeated NetworkservicesBetaTcpRoute items = 1;
}

service NetworkservicesBetaTcpRouteService {
  rpc ApplyNetworkservicesBetaTcpRoute(ApplyNetworkservicesBetaTcpRouteRequest) returns (NetworkservicesBetaTcpRoute);
  rpc DeleteNetworkservicesBetaTcpRoute(DeleteNetworkservicesBetaTcpRouteRequest) returns (google.protobuf.Empty);
  rpc ListNetworkservicesBetaTcpRoute(ListNetworkservicesBetaTcpRouteRequest) returns (ListNetworkservicesBetaTcpRouteResponse);
}
